pep-0606.rst

```diff
@@ -0,0 +1,223 @@
+
+PEP 605 proposes to address it by regularly delivering 2 months worth of changes
+to a subset of Python's user base that opts in to running a rolling stream of
+beta releases (similar to running Windows Insider builds instead of the Windows
+retail release, or running Debian testing instead of Debian stable).
+
+
+Reducing the latency of feature delivery
+----------------------------------------
+
+When only stable releases are seeing significant user adoption, and there's a
+long period of time between stable releases, it creates an incredibly strong
+temptation for developers to push changes into stable releases before they're
+really ready for general use.
+
+PEP 602 proposes to address this problem by reducing the period of time
+between stable releases to 12 months rather than 18 months.
+
+PEP 605 proposes to address it by actively seeking to create a community of
+Python users that regularly install and use CPython beta releases, providing an
+incentive for core developers to start shipping changes earlier in the
+pre-release cycle, in order to obtain feedback before the feature gets locked
+down in a stable release.
+
+
+Aligning the release cadence with the calendar year
+---------------------------------------------------
+
+While the current release cadence is nominally 18-24 months, in practice it has
+consistently been towards the 18 month end of that range. This means that the
+target dates for pre-releases and final releases move around from release to
+release, and the only way to remember them is to either look at the release PEP,
+or else to add those dates to your calendar. This is annoying for both
+individual volunteers and for corporate contributors, and also complicates
+alignment with events like PyCon US (typically April/May) and the now-annual
+core development sprints (typically in September).
+
+PEP 602 proposes to address this problem by publishing a new release in October
+every year, and basing the pre-release calendar for each year off that.
+
+PEP 605 proposes to address this problem by alternating between release years
+(where a new stable release is published in August), and non-release years
+(where only maintenance releases and new rolling beta releases are published).
+
+
+Improving the pre-release design feedback cycle
+-----------------------------------------------
+
+One of the challenges of designing changes to the core interpreter and standard
+library APIs is that the user base in a position to provide feedback on
+nightly builds and the current pre-releases is relatively limited. This means
+that much user feedback isn't received until after an API design has already
+shipped in a full X.Y.0 release.
+
+If the API is a regular API, then deprecation cycles mean that it may take
+literally years to correct any design mistakes identified at that point.
+Marking APIs as provisional offers a way to avoid that constraint, but actually
+taking advantage of that freedom causes other problems.
```

Comment:
Guido also stated that we won't do provisional APIs anymore, after the typing and asyncio experience.

I haven't heard of the SC rescinding that yet, and obviously a change to release process would justify revisiting that decision, but I think it's fine to say here that our current position is to not treat APIs as provisional.

Reply:
Is that something he said in person at the core dev sprints? I don't recall anyone writing that down anywhere.

Reply:
I implied this a bit more heavily by putting a "nominally" in the sentence.

Reply:
I think it was actually sprints two years ago? But yeah, nobody wrote it down...